### PR TITLE
Add Fuzzing

### PR DESCRIPTION
**Pull request by cpuu**
**[https://github.com/cpuu/awesome-fuzzing/](https://github.com/cpuu/awesome-fuzzing/)**

**[Fuzzing for security testing : related papers, workshop materials, video, tools and so on]**

I revewed : 
https://github.com/sindresorhus/awesome/pull/1493#issuecomment-452150802
https://github.com/sindresorhus/awesome/pull/1366#issuecomment-452151816

### By submitting this pull request I confirm I've read and complied with the below requirements 🖖

**Please read it multiple times. I spent a lot of time on these guidelines and most people miss a lot.**

## Requirements for your pull request

- **You have to review at least 2 other [open pull requests](https://github.com/sindresorhus/awesome/pulls?q=is%3Apr+is%3Aopen).** Try to prioritize unreviewed PRs, but you can also add more comments to reviewed PRs. Go through the below list when reviewing. This requirement is meant to help make the Awesome project self-sustaining. Comment here which PRs you reviewed. You're expected to put a good effort into this and to be thorough. Look at previous PR reviews for inspiration.
- I have read and understood the [instructions for creating a list](https://github.com/sindresorhus/awesome/blob/master/create-list.md).
- This pull request has a descriptive title in the format `Add Name of List` (Example: `Add Swift`), not `Update readme.md` or `Add awesome list`.
- The entry in the Awesome list should:
	- Include a short description about the project/theme of the list. **It should not describe the list itself.**<br>Example: `- [Fish](…) - User-friendly shell.`, not `- [Fish](…) - Resources for Fish.`.
	- Be added at the bottom of the appropriate category.
- The list I'm submitting complies with the below requirements.

## Requirements for your Awesome list

- **Has been around for at least 30 days.**<br>That means 30 days from either the first real commit or when it was open-sourced. Whatever is most recent.
- It's the result of hard work and the best I could possibly produce.
- Non-generated Markdown file in a GitHub repo.
- **Includes a succinct description of the project/theme at the top of the readme.** [(Example)](https://github.com/willempienaar/awesome-quantified-self)
- The repo should have `awesome-list` & `awesome` as [GitHub topics](https://help.github.com/articles/about-topics). I encourage you to add more relevant topics.
- Not a duplicate.
- Only has awesome items. Awesome lists are curations of the best, not everything.
- Includes a project logo/illustration whenever possible.
	- Either centered, fullwidth, or placed at the top-right of the readme. [(Example)](https://github.com/sindresorhus/awesome-electron)
	- The image should link to the project website or any relevant website.
	- The image should be high-DPI. Set it to maximum half the width of the original image.
- Entries have a description, unless the title is descriptive enough by itself. It rarely is though.
- Includes the [Awesome badge](https://github.com/sindresorhus/awesome/blob/master/awesome.md#awesome-badge).
	- Should be placed on the right side of the readme heading.
		- Can be placed centered if the list has a centered graphics header.
	- Should link back to this list.
- Has a Table of Contents section.
	- Should be named `Contents`, not `Table of Contents`.
	- Should be the first section in the list.
	- Should only have one level of [nested lists](https://commonmark.org/help/tutorial/10-nestedLists.html), preferably none.
- Has an [appropriate license](https://github.com/sindresorhus/awesome/blob/master/awesome.md#choose-an-appropriate-license).
	- That means something like CC0, **not a code licence like MIT, BSD, Apache, etc.**
	- [WTFPL](http://www.wtfpl.net) and [Unlicense](http://unlicense.org) are not acceptable licenses.
	- If you use a license badge, it should be SVG, not PNG.
- Has [contribution guidelines](https://github.com/sindresorhus/awesome/blob/master/awesome.md#include-contribution-guidelines).
	- The file should be named `contributing.md`. Casing is up to you.
- Has consistent formatting and proper spelling/grammar.
	- The link and description are separated by a dash. <br>Example: `- [AVA](…) - JavaScript test runner.`
	- The description starts with an uppercase character and ends with a period.
	- Consistent and correct naming. For example, `Node.js`, not `NodeJS` or `node.js`.
- Doesn't include a Travis badge.<br>You can still use Travis for list linting, but the badge has no value in the readme.

Go to the top and read it again.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for making an Awesome list! 🙌

It looks like you didn't read the [guidelines](https://github.com/sindresorhus/awesome/blob/master/pull_request_template.md) closely enough. I noticed multiple things that are not followed. Try going through the list point for point to ensure you follow it. I spent a lot of time creating the guidelines so I wouldn't have to comment on common mistakes, and rather spend my time improving Awesome.

**Comment by alexk111** [comment]
> The repo should have awesome-list & awesome as GitHub topics. I encourage you to add more relevant topics.

You didn't add the topics.

> Has a Table of Contents section.
> Should be named Contents, not Table of Contents

Currently = Table of Contents

> Has consistent formatting and proper spelling/grammar.
> The link and description are separated by a dash. 
> Example: - [AVA](…) - JavaScript test runner.
> The description starts with an uppercase character and ends with a period.

Tools section has inconsistent formatting:
- `Is a` instead of dash.
- Descriptions don't end with a period.
- No description.

**Comment by cpuu** [comment]
Thanks to @alexk111 ,  I have correctied what you mentioned.
@sindresorhus  , May I ask you to review again?

File: readme.md [hunk 1]
```diff
@@ -446,6 +446,7 @@
 - [Vehicle Security and Car Hacking](https://github.com/jaredthecoder/awesome-vehicle-security#readme)
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
+- [Fuzzing](https://github.com/cpuu/awesome-fuzzing#readme) - Fuzzing for security testing.
```
Comment (sindresorhus) [inline readme.md]: > The entry in the Awesome list should:
* Include a short description about the project/theme of the list. **It should not describe the list itself.**

**Comment by sindresorhus** [comment]
> fuzzing is the action of running a Program Under Test (PUT) with “fuzz inputs”. -Prof. Barton P Miller

This is not a good explanation of what fuzzing is.

**Comment by sindresorhus** [comment]
The repo description is out of sync with the readme.

**Comment by sindresorhus** [comment]
The readme title should be `Awesome Fuzzing`, not `Awesome Fuzzing & Fuzz Testing`

**Comment by sindresorhus** [comment]
```
❯ awesome-lint https://github.com/cpuu/awesome-fuzzing/
✖ Linting

  README.md:33:3
  ✖   15:1    ToC item "Platform" missing corresponding heading                                                                                                                                                                              remark-lint:awesome/toc
  ✖   33:3    Found reference to undefined definition                                                                                                                                                                                        remark-lint:no-undefined-references
  ✖   33:3    Invalid list item link                                                                                                                                                                                                         remark-lint:awesome/list-item
  ✖   38:3    Link to https://www.researchgate.net/profile/Cristiano_Giuffrida/publication/311886374_VUzzer_Application-aware_Evolutionary_Fuzzing/links/585f3f5608aebf17d38d5678/VUzzer-Application-aware-Evolutionary-Fuzzing.pdf is dead  remark-lint:no-dead-urls
  ✖   47:3    Link to https://www.andrew.cmu.edu/user/rkersten/publications/ccs17-kersten.pdf is dead                                                                                                                                        remark-lint:no-dead-urls
  ✖   49:3    Link to https://www.comp.nus.edu.sg/~mboehme/paper/CCS17.pdf is dead                                                                                                                                                           remark-lint:no-dead-urls
  ✖   51:112  List item link and description must be separated with a dash                                                                                                                                                                   remark-lint:awesome/list-item
  ✖   51:115  Text "github" should be written as "GitHub"                                                                                                                                                                                    remark-lint:awesome/spell-check
  ✖  105:98   List item description must end with proper punctuation                                                                                                                                                                         remark-lint:awesome/list-item
  ✖  109:58   List item description must end with proper punctuation                                                                                                                                                                         remark-lint:awesome/list-item
  ✖  110:49   List item description must end with proper punctuation                                                                                                                                                                         remark-lint:awesome/list-item
  ✖  111:136  List item description must end with proper punctuation                                                                                                                                                                         remark-lint:awesome/list-item
  ✖  113:58   List item description must start with valid casing                                                                                                                                                                             remark-lint:awesome/list-item
  ✖  124:103  List item link and description must be separated with a dash                                                                                                                                                                   remark-lint:awesome/list-item

  14 errors
```

https://github.com/sindresorhus/awesome-lint

**Comment by cpuu** [comment]
How great is awesome-lint tool!!
I modified according to the results.
```
cpuu@cpuu:~$ awesome-lint https://github.com/cpuu/awesome-fuzzing/
✔ Linting
```

File: readme.md [hunk 2]
```diff
@@ -446,6 +446,7 @@
 - [Vehicle Security and Car Hacking](https://github.com/jaredthecoder/awesome-vehicle-security#readme)
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
+- [Fuzzing](https://github.com/cpuu/awesome-fuzzing#readme) - Fuzzing is widely used by both software security and by quality assurance.
```
Comment (sindresorhus) [inline readme.md]: Ok, so it's widely used, but that's useless info by itself. It doesn't explain **what fuzzing is**.

**Comment by sindresorhus** [comment]
You have not fixed all my points. Please read **all** my comments.

**Comment by sindresorhus** [comment]
Linkify `Fuzzing` in `Fuzzing or fuzz testing is an automated software testing technique that` to https://en.wikipedia.org/wiki/Fuzzing

**Comment by sindresorhus** [comment]
For the entries that link to a paper, it would be useful to add a description with a short summary of what the paper is about.

**Comment by sindresorhus** [comment]
`python` => `Python`

**Comment by sindresorhus** [comment]
> To achieve a well-defined scope, I have chosen to include all publications on fuzzing in the last proceedings of 4 major security conferences and others from Jan 2008 to Feb 2019.

Awesome lists should be a curation of the best, not everything.

**Comment by sindresorhus** [comment]
The sub-sections in the `Paper` section could use a description on what they are. For example, not clear what `USENIX Security` is.

**Comment by sindresorhus** [comment]
`Randoop : Feedback-directed` => `Randoop: Feedback-directed`

**Comment by cpuu** [comment]
> > To achieve a well-defined scope, I have chosen to include all publications on fuzzing in the last proceedings of 4 major security conferences and others from Jan 2008 to Feb 2019.
> 
> Awesome lists should be a curation of the best, not everything.

I selected only the best of the latest research trends related to fuzzing.
It is a mistake to write 'all'. it's a 'The only and the best of the best' list.
There are hundreds of security-related conferences. Among them, 4 were selected as the best societies. In those of papers, only the best papers on 'fuzzing' were selected.

**Comment by cpuu** [comment]
> `Randoop : Feedback-directed` => `Randoop: Feedback-directed`

I confirmed that Randoop is an alias, and the official name of the paper is "Feedback-directed ...", so I modified it accordingly.

**Comment by cpuu** [comment]
> The sub-sections in the `Paper` section could use a description on what they are. For example, not clear what `USENIX Security` is.

I included description of each conferences which are top 4 security society

**Comment by cpuu** [comment]
> For the entries that link to a paper, it would be useful to add a description with a short summary of what the paper is about.

It is a very good opinion and I hope to do so. But it will take a very long time and I will proceed gradually.

File: readme.md (hunk 2 above)
Comment (cpuu) [reply readme.md]: I know what you mean. I changed !
However, there are a lot of items that are not explained enough in the awesome list.
Is this necessary? If not, I will skip it.

File: readme.md [hunk 3]
```diff
@@ -446,6 +446,7 @@
 - [Vehicle Security and Car Hacking](https://github.com/jaredthecoder/awesome-vehicle-security#readme)
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
+- [Fuzzing](https://github.com/cpuu/awesome-fuzzing#readme) - Fuzzing(for software security) related papers and their implementation codes, tools reported at the top security conference.
```
Comment (sindresorhus) [inline readme.md]: This still doesn't explain what **fuzzing** is.

Comment (cpuu) [reply readme.md]: I added a sentence describing what fuzzing is, (a kind of software security test method).
More details are already included in the link.

**Comment by cpuu** [comment]
I added a sentence describing what fuzzing is, (a kind of software security test method).
More details are already included in the link.
and I merged branch

File: readme.md [hunk 4]
```diff
@@ -453,6 +453,7 @@
 - [Web Security](https://github.com/qazbnm456/awesome-web-security#readme) - Security of web apps & services.
 - [Lockpicking](https://github.com/meitar/awesome-lockpicking#readme) - The art of unlocking a lock by manipulating its components without the key.
 - [Cybersecurity Blue Team](https://github.com/meitar/awesome-cybersecurity-blueteam#readme) - Groups of individuals who identify security flaws in information technology systems.
+- [Fuzzing](https://github.com/cpuu/awesome-fuzzing#readme) - Fuzzing is a type of software security testing, and this link describes fuzzing related awesome papers and their implementation codes, tools reported at the top security conference.
```
Comment (sindresorhus) [inline readme.md]: This is still not acceptable description. I don't really know how to explain it clearer. Try to look at the other descriptions and see the pattern.

Comment (janisozaur) [inline readme.md]:
```suggestion
- [Fuzzing](https://github.com/cpuu/awesome-fuzzing#readme) - Tools and techniques for testing applications by feeding them pseudo-randomly generated input data.
```

**Comment by sindresorhus** [comment]
I'm reluctantly merging this. I hope you do a better effort maintaining the list than you did with this PR.

**Pull request merged**